### PR TITLE
feat: run clippy + BlueBuildTrait

### PR DESCRIPTION
**Pull request by n16hth4wkk**
- Run clippy and add clippy warnings for better code cleanness
- Condense commands to a trait and shove them into a folder together

### CODE REVIEW

File: src/commands/mod.rs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#[cfg(feature = "init")]
+pub mod init;
+
+pub mod build;
+pub mod local;
+pub mod template;
+
+pub trait BlueBuildCommand {
+    fn run(&mut self);
```
Comment (gmpinder) [inline src/commands/mod.rs:9]: I'm thinking, maybe create a default implementation that checks if try_run errors, prints the error, and exits the process. That'll help reduce more repeated code

File: src/lib.rs [hunk 2]
```diff
@@ -1,16 +1,12 @@
 //! The root library for blue-build.
-
+#![warn(clippy::correctness, clippy::suspicious, clippy::perf, clippy::style)]
 #![doc(
     html_logo_url = "https://gitlab.com/wunker-bunker/blue-build/-/raw/main/logos/BlueBuild-logo.png"
 )]
 #![doc = include_str!("../README.md")]
 #![forbid(unsafe_code)]
 #![allow(unused_imports)]
+#![allow(clippy::module_name_repetitions)]
```
Comment (gmpinder) [inline src/lib.rs:9]: What does this end up doing?

**Review by gmpinder — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (n16hth4wkk) [reply src/lib.rs:9]: clippy throws warnings if the sturct or trait has the same ending as the file name.

ex: ContainerFileTemplate -> template.rs

I find this pretty annoying lol I like to defined an ending no matter the file

File: src/commands/mod.rs (hunk 1 above)
Comment (n16hth4wkk) [reply src/commands/mod.rs:9]: oh yeah this will help alot

File: src/lib.rs (hunk 2 above)
Comment (gmpinder) [reply src/lib.rs:9]: Ok yeah makes sense

File: src/commands/build.rs [hunk 3]
```diff
@@ -493,6 +500,9 @@ impl BuildCommand {
         Ok(image_name)
     }
 
+    /// # Errors
+    ///
+    /// Will return `Err` if the build fails.
```
Comment (gmpinder) [inline src/commands/build.rs:505]: Was this part of the pedantic mode?

**Review by gmpinder — APPROVED** [review]

**Pull request merged**